### PR TITLE
Recyclebin improvements

### LINKED ISSUE #1088: Moving a file will move the original file to recycle bin
**Describe the bug**
When you cut a file and paste it, Files UWP will move the original file to the recycle bin.

**To Reproduce**
1. Cut a file
2. Paste it somewhere else
3. Go to recycle bin
4. You'll see the original file

**Expected behavior**
The original file should be permanently deleted.

**Desktop (please complete the following information):**
 - OS Version: Windows 10 Pro Insider Preview Build 19645.1
 - App version: v0.10.2

### PR DESCRIPTION
**Pull request by gave92**
Implement some of the changes suggested here #1089
Changes:
- Fix #1088, #1181, #1199
- Auto refresh on recyclebin changes
- Fill info in Properties window
- Customized recyclebin context menu
- Recycle bin always at the bottom of sidebar items
- Added confirmation dialog on emptying recyclebin

### CODE REVIEW

File: Files/View Models/SelectedItemsPropertiesViewModel.cs [hunk 1]
```diff
@@ -502,7 +502,7 @@ public SelectedItemsPropertiesViewModel(ListedItem item)
             {
                 ItemName = Item.ItemName;
                 ItemType = Item.ItemType;
-                ItemPath = Path.GetDirectoryName(Item.ItemPath);
+                ItemPath = Path.IsPathRooted(Item?.ItemPath) ? Path.GetDirectoryName(Item?.ItemPath) : Item?.ItemPath;
```
Comment (tsvietOK) [inline Files/View Models/SelectedItemsPropertiesViewModel.cs:505]:
```suggestion
                ItemPath = Path.IsPathRooted(Item.ItemPath) ? Path.GetDirectoryName(Item.ItemPath) : Item.ItemPath;
```

File: Files/UserControls/YourHome.xaml.cs [hunk 2]
```diff
@@ -185,7 +185,7 @@ private async void RecentsView_ItemClick(object sender, ItemClickEventArgs e)
             }
             catch (COMException)
             {
-                await DialogDisplayHelper.ShowDialog(ResourceController.GetTranslation("DriveUnpluggedDialog.Title"), ResourceController.GetTranslation("DriveUnpluggedDialog.Text"));
+                await DialogDisplayHelper.ShowDialog(ResourceController.GetTranslation("DriveUnpluggedDialog/Title"), ResourceController.GetTranslation("DriveUnpluggedDialog/Text"));
```
Comment (tsvietOK) [inline Files/UserControls/YourHome.xaml.cs:188]:
```csharp
                await DialogDisplayHelper.ShowDialog(
                    ResourceController.GetTranslation("DriveUnpluggedDialog/Title"),
                    ResourceController.GetTranslation("DriveUnpluggedDialog/Text"));
```

File: Files/View Models/ItemViewModel.cs [hunk 3]
```diff
@@ -813,13 +814,13 @@ public async Task EnumerateItemsFromStandardFolder(string path)
             }
             catch (FileNotFoundException)
             {
-                await DialogDisplayHelper.ShowDialog(ResourceController.GetTranslation("FolderNotFoundDialog.Title"), ResourceController.GetTranslation("FolderNotFoundDialog.Text"));
+                await DialogDisplayHelper.ShowDialog(ResourceController.GetTranslation("FolderNotFoundDialog/Title"), ResourceController.GetTranslation("FolderNotFoundDialog/Text"));
```
Comment (tsvietOK) [inline Files/View Models/ItemViewModel.cs:817]:
```csharp
                await DialogDisplayHelper.ShowDialog(
                    ResourceController.GetTranslation("FolderNotFoundDialog/Title"),
                    ResourceController.GetTranslation("FolderNotFoundDialog/Text"));
```

File: Common/ShellFileItem.cs [hunk 4]
```diff
@@ -10,14 +10,14 @@ public class ShellFileItem
...
-        public long FileSizeBytes;
+        public ulong FileSizeBytes;
         public string FileType;
 
         public ShellFileItem()
         {
         }
 
-        public ShellFileItem(bool isFolder, string recyclePath, string fileName, string filePath, DateTime recycleDate, string fileSize, long fileSizeBytes, string fileType)
+        public ShellFileItem(bool isFolder, string recyclePath, string fileName, string filePath, DateTime recycleDate, string fileSize, ulong fileSizeBytes, string fileType)
```
Comment (tsvietOK) [inline Common/ShellFileItem.cs:20]:
```csharp
        public ShellFileItem(
            bool isFolder, string recyclePath, string fileName, string filePath,
            DateTime recycleDate, string fileSize, ulong fileSizeBytes, string fileType)
```

**Comment by tsvietOK** [comment]
Issue #1199 is not fixed. Still getting the same exception in debug_fulltrust.txt

File: Files/Interacts/Interaction.cs [hunk 5]
```diff
@@ -654,7 +654,7 @@ public async void DeleteItem_Click(object sender, RoutedEventArgs e)
             }
             catch (IOException)
             {
-                if (await DialogDisplayHelper.ShowDialog(ResourceController.GetTranslation("FileInUseDeleteDialog.Title"), ResourceController.GetTranslation("FileInUseDeleteDialog.Text"), ResourceController.GetTranslation("FileInUseDeleteDialog.PrimaryButtonText"), ResourceController.GetTranslation("FileInUseDeleteDialog.SecondaryButtonText")))
+                if (await DialogDisplayHelper.ShowDialog(ResourceController.GetTranslation("FileInUseDeleteDialog/Title"), ResourceController.GetTranslation("FileInUseDeleteDialog/Text"), ResourceController.GetTranslation("FileInUseDeleteDialog/PrimaryButtonText"), ResourceController.GetTranslation("FileInUseDeleteDialog/SecondaryButtonText")))
```
[the remark this answers is not in the document]
Comment (gave92) [reply Files/Interacts/Interaction.cs:657]: Should this be done in this PR? All the others dialogs are written on one line.

**Comment by gave92** [comment]
@tsvietOK Can't reproduce that exception anymore, mind telling me on which line is thrown?

File: Files/Interacts/Interaction.cs (hunk 5 above)
[the remark this answers is not in the document]
Comment (tsvietOK) [reply Files/Interacts/Interaction.cs:657]: If you changing this line, so you could align it.

**Comment by tsvietOK** [comment]
I can't tell you where the exception, but that's what i get:
image

**Comment by gave92** [comment]
To get the exception line you could:
- launch Files Dev from start menu
- in visual studio:
    - click Debug menu
    - click Attach to process..
    - select "FilesFullTrust.exe"
- go to recyclebin in Files Dev and get exception

In the picture I'm seeing a "NullReferenceException", in issue #1199 it was an "InvalidCastException". Is the debug_fulltrust.txt still the same?

If it doesn't work I can make a version with some additional debug logs.

**Comment by tsvietOK** [comment]
@gave92 
image

**Comment by gave92** [comment]
@tsvietOK Could you confirm whether the issue you reported is still present in the latest commit? Thanks!

**Comment by tsvietOK** [comment]
@gave92 Issue fixed, good job!

**Comment by yaichenbaum** [comment]
@gave92 Can you merge in `master`?

**Pull request merged**